modules/globebrowsing/globes/renderableglobe.cpp

```diff
@@ -258,8 +245,32 @@ const std::shared_ptr<const Camera> RenderableGlobe::savedCamera() const {
     return _savedCamera;
 }
 
-double RenderableGlobe::interactionDepthBelowEllipsoid() {
-    return _interactionDepthBelowEllipsoid;
+SurfacePositionHandle RenderableGlobe::calculateSurfacePositionHandle(
+    glm::dvec3 targetModelSpace) 
+{
+    glm::dvec3 centerToEllipsoidSurface = _ellipsoid.geodeticSurfaceProjection(targetModelSpace);
```

Comment:
I made the function `calculateSurfacePositionHandle` non const because it uses sampling of height layers which in turn might initiate new reading calls to tile providers etc. Those cannot be const. Is it crucial that this is const? It would probably require some rewriting in how layers are sampled. I don't really want to spend time on that.